html5lib.parse should allow bytes

This example from the [html5lib docs](https://html5lib.readthedocs.io/en/latest/#usage):

> Simple usage follows this pattern:
>
> ```python
> import html5lib
> with open("mydocument.html", "rb") as f:
>     document = html5lib.parse(f)
> ```

fails to type-check:

```
demo2.py:3: error: Argument 1 to "parse" has incompatible type "BufferedReader"; expected "str | SupportsRead[str]"  [arg-type]
demo2.py:3: note: Following member(s) of "BufferedReader" have conflicts:
demo2.py:3: note:     Expected:
demo2.py:3: note:         def read(self, int = ..., /) -> str
demo2.py:3: note:     Got:
demo2.py:3: note:         def read(self, int | None = ..., /) -> bytes
Found 1 error in 1 file (checked 1 source file)
```

The parser takes any of `bytes`, `str`, binary files and text files -- see [the source](https://github.com/html5lib/html5lib-python/blob/fd4f032bc090d44fb11a84b352dad7cbee0a4745/html5lib/_inputstream.py#L125-L145). But the typing allows text strings only.

Note that html5lib was first released in 2013 and still supports Python 2; mentions of “string” in the code/docs might still refer to bytes.

--- a/stubs/html5lib/html5lib/_inputstream.pyi
+++ b/stubs/html5lib/html5lib/_inputstream.pyi
@@ -11,7 +11,7 @@ class _Encoding(Protocol):
 
 _UnicodeInputStream: TypeAlias = str | SupportsRead[str]
 _BinaryInputStream: TypeAlias = bytes | SupportsRead[bytes]
-_InputStream: TypeAlias = _UnicodeInputStream  # noqa: Y047  # used in other files
+_InputStream: TypeAlias = _UnicodeInputStream | _BinaryInputStream  # noqa: Y047  # used in other files
 
 spaceCharactersBytes: Any
 asciiLettersBytes: Any

